internal/sql/sql.go

```diff
@@ -46,6 +46,8 @@ type (
 		// Exec executes a query without returning any rows. The args are for
 		// any placeholder parameters in the query.
 		Exec(ctx context.Context, query string, args ...any) (sql.Result, error)
+		// LoggedRows wraps a *sql.Rows with a logger and a long query duration.
+		LoggedRows(rows *sql.Rows) *LoggedRows
```

Comment:
Instead of this, I would update `func (ls *LoggedStmt) Query(ctx context.Context, args ...any) (*sql.Rows, error) {` to return logged rows as it should anyway.

e.g.
`func (ls *LoggedStmt) QueryRow(ctx context.Context, args ...any) *LoggedRow {` already returns a logged row.